src/common/task_system/task_scheduler.cpp

```diff
@@ -33,6 +36,7 @@ void TaskScheduler::scheduleTaskAndWaitOrError(
         scheduleTaskAndWaitOrError(dependency, context);
     }
     auto scheduledTask = scheduleTask(task);
+    cv.notify_all();
```

Comment:
We should notify inside `scheduleTask`, not here.

Reply:
scheduleTask takes the mutex. I think notifying with mutex causes more contention

Reply:
Then we should `unlock()` before notifying, right?

Reply:
But https://en.cppreference.com/w/cpp/thread/unique_lock/unlock exists, right?

Reply:
Yeah but I don’t think unlocking manually, then notifying, then return makes much of a difference from returning, then unlocking automatically, the notifying since returning a pointer should take almost no time.

Reply:
My point is more that `scheduleTask()` is public, and should absolutely be waking up the worker threads, or the method is wrong

Reply:
Actually I think scheduleTask could be made private instead. I did a quick search and It seems it is only being called from scheduleTaskAndWaitOrError and scheduleTaskAndWaitOrError is called by processor.